src/cdk-lib/bedrock/application-inference-profile.ts

```diff
@@ -0,0 +1,210 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+import { IResource, Resource, CfnTag, Arn, ArnFormat } from 'aws-cdk-lib';
+import * as bedrock from 'aws-cdk-lib/aws-bedrock';
+import { Construct } from 'constructs';
+
+/******************************************************************************
+ *                              COMMON
+ *****************************************************************************/
+/**
+ * Represents a ApplicationInferenceProfile, either created with CDK or imported.
+ */
+export interface IApplicationInferenceProfile extends IResource {
```

Comment:
I think it would be good idea to define a common interface for both ApplicationInferenceProfiles as CrossRegionInferenceProfiles. Something along the lines of 
```ts

export enum InferenceProfileType {
  SYSTEM_DEFINED = "SYSTEM_DEFINED".
  APPLICATION="APPLICATION"
}

export interface IInferenceProfile extends IResource {
  /**
   * The ARN of the application inference profile.
   */
  readonly inferenceProfileArn: string;
  /**
   * The unique identifier of the inference profile.
   */
  readonly inferenceProfileId: string;
  /**
   * The type of inference profile.
   */
  readonly type: InferenceProfileType;
}
```

Reply:
Yes, after discussion it is great, let's do it. It will also help to have model source through props strongly typed instead of a string

Reply:
Resolving as it is part of latest updates